src/base/rss/private/rss_parser.cpp

```diff
@@ -206,11 +207,23 @@ namespace
     }
 }
 
-using namespace Rss::Private;
+using namespace RSS::Private;
+
+Parser::Parser(QString lastBuildDate)
+{
+    qRegisterMetaType<ParsingResult>();
+    m_result.lastBuildDate = lastBuildDate;
+}
 
-// read and create items from a rss document
 void Parser::parse(const QByteArray &feedData)
 {
+    QMetaObject::invokeMethod(this, "parse_impl", Qt::QueuedConnection
```

Comment:
Why for the main thread? Queued slots are called in thread of its object. In case the parser is in another thread it will be called there. We use parsers in another thread (`Session::workingThread()`).